packages/client/hmi-client/src/components/workflow/tera-create-workflow-modal.vue

```diff
@@ -0,0 +1,109 @@
+					</div>
+				</aside>
+				<main class="col-10 flex flex-column gap-3">
+					<component
+						v-if="getScenario()"
+						:is="getScenario()?.component"
+						:state="getScenario()?.template"
+						@update-state="onUpdateState"
+					/>
+				</main>
+			</div>
+		</template>
+		<template #footer>
+			<Button label="Create" size="large" @click="saveWorkflow" />
+			<Button label="Close" class="p-button-secondary" size="large" outlined @click="emit('close-modal')" />
+		</template>
+	</tera-modal>
+</template>
+
+<script setup lang="ts">
+import TeraModal from '@/components/widgets/tera-modal.vue';
+import Button from 'primevue/button';
+import { ref } from 'vue';
+import type { Component } from 'vue';
+import RadioButton from 'primevue/radiobutton';
+import { BaseScenarioTemplate } from '@/components/workflow/scenario-templates/scenario-template';
+import { createWorkflow } from '@/services/workflow';
+import { AssetType } from '@/types/Types';
+import { useProjects } from '@/composables/project';
+import router from '@/router';
+import { RouteName } from '@/router/routes';
+import TeraBlankCanvasTemplate from '@/components/workflow/scenario-templates/blank-canvas/tera-blank-canvas-template.vue';
+import TeraSituationalAwarenessTemplate from '@/components/workflow/scenario-templates/situational-awareness/tera-situational-awareness-template.vue';
+import { BlankCanvasScenarioTemplate } from '@/components/workflow/scenario-templates/blank-canvas/blank-canvas-template';
+import { SituationalAwarenessScenarioTemplate } from '@/components/workflow/scenario-templates/situational-awareness/situational-awareness-template';
+
+interface ScenarioItem {
+	template: BaseScenarioTemplate;
+	component: Component;
+}
+
+const scenarios = ref<ScenarioItem[]>([
+	{
+		template: new BlankCanvasScenarioTemplate(),
+		component: TeraBlankCanvasTemplate
+	},
+	{
+		template: new SituationalAwarenessScenarioTemplate(),
+		component: TeraSituationalAwarenessTemplate
+	}
+]);
+
+const emit = defineEmits(['close-modal']);
+
+const selectedTemplateId = ref<any>(scenarios.value[0].template.templateId);
+
+const saveWorkflow = async () => {
+	const scenario = getScenario();
+	if (!scenario) {
+		return;
+	}
+	const wf = scenario.template.createWorkflow();
+	const response = await createWorkflow(wf);
+
+	const projectId = useProjects().activeProject.value?.id;
+	await useProjects().addAsset(AssetType.Workflow, response.id, projectId);
+
+	await useProjects().refresh();
+
+	// redirect to the asset page
+	router.push({
+		name: RouteName.Project,
+		params: {
+			projectId,
+			pageType: AssetType.Workflow,
+			assetId: response.id
+		}
+	});
+
+	emit('close-modal');
+};
+
+const onUpdateState = (state: any) => {
+	const scenario = getScenario();
+	if (!scenario) {
+		return;
+	}
+	Object.assign(scenario.template, state);
+};
+
+const getScenario = () => scenarios.value.find((s) => s.template.templateId === selectedTemplateId.value);
```

Comment:
If we know it can never be null or undefined, let's just cast it so we don't get silly typescript warnings.

```
const getScenario = () =>  scenarios.find(...) as XYZ
```